.github/workflows/test.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
```

Comment:
Is this line needed?

Reply:
just in case we want to rerun the workflow manually once on github

Reply:
Yep, thought you had to add some details about it.